types/fibjs/declare/SandBox.d.ts

```diff
@@ -170,6 +210,44 @@ declare class Class_SandBox extends Class__object {
 	 */
 	require(id: string, base: string): any;
 
+	/**
+	 * 
+	 * @brief 对指定的 extname 添加 compiler, extname 不可为系统内置扩展名 (包括 {'.js', '.json', '.jsc', '.wasm'}), compiler 需返回有效的 javascript 脚本.
+	 * 
+	 * ```JavaScript
+	 * var vm = require('vm');
+	 * var sbox = new vm.SandBox({
+	 * });
+	 * 
+	 * // 编译 typescript 脚本为 js 并加载
+	 * sbox.setModuleCompiler('.ts', tsCompiler);
+	 * var mod_ts = sbox.require('./a.ts');
+	 * 
+	 * // 编译 coffee 脚本为 js 并加载
+	 * sbox.setModuleCompiler('.coffee', cafeCompiler);
+	 * var mod_coffee = sbox.require('./a.coffee');
+	 * 
+	 * // 编译 jsx 脚本为 js 并加载
+	 * sbox.setModuleCompiler('.jsx', reactCompiler);
+	 * var mod_react = sbox.require('./a.jsx');
+	 * 
+	 * // 编译 yml 脚本为自定义的内容(如 API 集合) 并加载
+	 * sbox.setModuleCompiler('.yml', yaml2Rest)
+	 * sbox.setModuleCompiler('.yaml', yaml2Rest)
+	 * 
+	 * // 编译 markdown 为自定义的内容(如 html 字符串或 XmlDocument 对象) 并加载
+	 * sbox.setModuleCompiler('.md', mdCompiler)
+	 * sbox.setModuleCompiler('.markdown', mdCompiler)
+	 * ```
+	 * 
+	 * @param extname 指定的 extname, 必须以 '.' 开头, 且为非系统内置扩展名
+	 * @param compiler 编译回调函数, 所有带 extname 的文件仅会 require 一次. 该回调函数格式为 `compiler(buf, requireInfo)`, buf 为读取到的文件 Buffer, requireInfo 结构为 `{filename: string}`.
+	 * 
+	 * 
+	 * 
+	 */
+	setModuleCompiler(extname: string, compiler: Function): void;
```

Comment:
Thanks for your advice to improving our types! In fact, we tried to generate more specific function-type param in our custom Object/module descriptor file format with `.idl` suffix (you can learn more details of it [here](https://github.com/fibjs/fibjs/blob/master/idl/us-en/global.idl)), it was born to generate header file and virtual class base of fibjs's C++ source code(like [this one](https://github.com/fibjs/fibjs/blob/master/fibjs/include/ifs/global.h) you could found words *This file was automatically generated using idlc.js*). So, sometimes definitions of `.idl` cannot  always specify one explicit one typescript's function type(or say, interface), in this case, we can just mark it as 'Function' rather than specific interface :(

We are considering add some type-notation of `.idl` to specify function-type/interface, but that's not one simple determination, we must be so careful that we can change it right now.

On the other hand, my English is not very well, if some words above is weird, never care about that : )